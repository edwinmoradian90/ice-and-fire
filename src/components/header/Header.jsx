import React, { useState, useEffect } from 'react'
import { StyledLink, Title } from '../../utils/styledComponents/main'

function Header(props) {
    const { category } = props
    const [title, setTitle] = useState('Ice and Fire')
    const [categoryName, setCategoryName] = useState('')
    const [showSearch, setShowSearch] = useState(true)

    useEffect(() => {
        if (category) {
            setCategoryName(category)
        }
    }, [category])

    return (
        <div className="header">
            <StyledLink to='/'>
                <Title className="appTitle">{title}</Title>
            </StyledLink>
        </div>
    )
}

export default Header