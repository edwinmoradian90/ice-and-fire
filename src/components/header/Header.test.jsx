import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { BrowserRouter as Router } from 'react-router-dom';

test('Returns snapshot and application text', () => {
    const component = renderer.create(
        <Router>
            <Header />
        </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(tree.children[0].children[0].children[0]).toContain('Ice and Fire');
});