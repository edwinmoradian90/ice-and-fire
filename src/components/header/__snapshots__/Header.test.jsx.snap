// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns snapshot and application text 1`] = `
<div
  className="header"
>
  <a
    className="sc-AxjAm irMvbk"
    href="/"
    onClick={[Function]}
  >
    <h1
      className="sc-AxhCb kYEhcE appTitle"
    >
      Ice and Fire
    </h1>
  </a>
</div>
`;
